src/ucs/datastruct/lru.h

```diff
@@ -116,12 +118,15 @@ static UCS_F_ALWAYS_INLINE void ucs_lru_put(ucs_lru_h lru, const void *key)
 /**
  * @brief Get a list of all elements in the cache.
  *
- * @param [in]  lru   Handle to the LRU cache.
- * @param [out] array Pointer to an allocated ucs_array to store elements.
- *                    Array must be at least of 'lru->capacity' capacity.
+ * @param [in]  lru     Handle to the LRU cache.
+ * @param [in]  cb      Callback function to be called for each element in the
+ *                      cache.
+ * @param [in]  opaque  Pointer to user data object. Will be passed to the
+ *                      callback function.
  *
  * @return UCS_OK if successful, or error code otherwise.
  */
-ucs_status_t ucs_lru_get_all(const ucs_lru_h lru, ucs_lru_array_t *array);
+ucs_status_t
+ucs_lru_get_all(const ucs_lru_h lru, ucs_lru_get_cb_t cb, void *opaque);

```

Comment:
The name of the func is not indicative anymore. It doesn't "get" in the sense that - it doesn't **return**  the whole contents, but rather iterates over the LRU and "visits" each element with a user-defined callback.